src/Linkify.jsx

```diff
@@ -13,19 +38,62 @@ class Linkify extends React.Component {
     component: React.PropTypes.any,
     properties: React.PropTypes.object,
     urlRegex: React.PropTypes.object,
-    emailRegex: React.PropTypes.object
+    emailRegex: React.PropTypes.object,
+    handlers: React.PropTypes.arrayOf(
+      React.PropTypes.shape({
+        prefix: React.PropTypes.string.isRequired,
+        validate: React.PropTypes.func.isRequired,
+        normalize: React.PropTypes.func.isRequired
+      })
+    ),
+    fuzzyLink: React.PropTypes.bool,
+    fuzzyIP: React.PropTypes.bool,
+    fuzzyEmail: React.PropTypes.bool
   }
 
   static defaultProps = {
     className: 'Linkify',
     component: 'a',
     properties: {},
+    handlers: []
+  }
+
+  componentDidMount() {
+    this.addCustomHandlers();
+  }
+
+  componentDidUpdate(nextProps) {
+    this.addCustomHandlers();
+  }
+
+  addCustomHandlers() {
+    const { handlers } = this.props;
+
+    this.linkify = this.linkify || new LinkifyIt();
+    this.linkify.tlds(tlds);
+
+    // add global customizations
+    for (let type in globalCustomizations) {
+      globalCustomizations[type].forEach(c => this.linkify[type](...c))
+    }
+
+    // add instance customizations
+    (handlers || []).forEach((handler) => {
+      this.linkify.add(handler.prefix, {
```

Comment:
What happens to handlers that are removed in the new props? Removed handlers should be set to `null` to disable them (see https://github.com/markdown-it/linkify-it#addschema-definition).